server-spi-private/src/main/java/org/keycloak/models/utils/KeycloakModelUtils.java

```diff
@@ -922,6 +919,14 @@ public static boolean isFlowUsed(RealmModel realm, AuthenticationFlowModel model
         if ((realmFlow = realm.getDockerAuthenticationFlow()) != null && realmFlow.getId().equals(model.getId())) return true;
         if ((realmFlow = realm.getFirstBrokerLoginFlow()) != null && realmFlow.getId().equals(model.getId())) return true;
 
+        boolean overrideByClient = realm.getClientsStream()
```

Comment:
This is performance killer in case that realm has many clients.

There are those methods introduced https://github.com/keycloak/keycloak/pull/21058/files in some old PR. Could you please rather use those instead of `realms.getClientsStream()` with the programmatic filtering?